### DISCUSSION TITLE
Battery with a different sensor for charge and discharge

### DISCUSSION THREAD

**Discussion by SNAPE15** [slipx06/sunsynk-power-flow-card#333] [Q&A] [answered]
Is it possible to use 2 sensors for the battery flow, 1 showing charge and 1 showing discharge? As my inverter doesnt give a single sensor with negative and positive values.
Love the card, slowly working out how to use it with a Sungrow Inverter.

**Comment by slipx06** [comment, marked as answer]
Hi. You need to create a template sensor that does this. I think it may have been asked here before but if not you can also find examples on the HA forum. Something like the below

```
sensor:
  - platform: template
    sensors:
      battery_power_combined:
        friendly_name: "Battery Power Combined"
        unit_of_measurement: "W"
        device_class: "Power"
        value_template: >-
          {% set charge_power = states('sensor.battery_charge_power') | float %}
          {% set discharge_power = states('sensor.battery_discharge_power') | float %}
          {% if charge_power > 0 %}
            {{ -charge_power }}
          {% elif discharge_power > 0 %}
            {{ discharge_power }}
          {% else %}
            0
          {% endif %}
```

**Reply by SNAPE15** [reply]
Thanks. That one didnt work for me but i was able to create a helper to perform the task.

Another question if you don't mind?

Is there a way to make the energy_cost_sell figure display even when there is no exporting happening? Currently it only displays when the system is exporting.

**Reply by slipx06** [reply]
The easy way is to use the same sensor (sell cost) for energy_cost_buy

**Reply by SNAPE15** [reply]
Thank you! Was the easy option.
The best power/energy flow card i've used. Got most of it working very well with my Sungrow Inverter/battery system.
